File: algorithms/sortings/selectionSort.js
// Selection Sort - Quadratic Sorting
function selectionSort(arr) {
  for (let i = 0; i < arr.length - 1; i++) {
    let select = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[j] <= arr[select]) {
        select = j;
      }
    }
    swap(arr, select, i);
  }
}

function swap(arr, i, j) {
  let temp = arr[i];
  arr[i] = arr[j];
  arr[j] = temp;
}

const arr = [4, 2, 8, 12, 1, 9];
selectionSort(arr);
console.log(arr);

File: algorithms/sortings/selection_sort.py
# [4, 3, 2, 10, 5, 8]

def selection_sort(arr):
  for i in range(0, len(arr)):
    select = i
    for j in range(i + 1, len(arr)):
      if arr[j] < arr[select]:
        select = j
    arr[i], arr[select] = arr[select], arr[i]

# unsorted array
arr = [4, 3, 2, 10, 5, 8]
selection_sort(arr)
# sorted array
print(arr)